cli/tools/test.rs

```diff
@@ -314,9 +319,24 @@ impl TestReporter for PrettyTestReporter {
   }
 
   fn report_output(&mut self, output: &TestOutput) {
-    if self.echo_output {
-      match output {
-        TestOutput::Console(line) => println!("{}", line),
+    if !self.echo_output {
+      return;
+    }
+
+    if !self.did_have_user_output {
+      self.did_have_user_output = true;
+      println!();
+      println!("{}", colors::gray("-------output:-------"));
+    }
+    match output {
+      TestOutput::PrintStdout(line) | TestOutput::PrintStderr(line) => {
+        print!("{}", line)
```

Comment:
This should probably be `print!` for `PrintStdout` and `eprint!` for `PrintStderr`